include/pinocchio/multibody/joint/joint-ellipsoid.hpp

```diff
@@ -0,0 +1,834 @@
+
+        // Third row: [0,0,0,0,0,1]^T · f = f.angular()[2]
+        res[2] = f.angular()[2];
+
+        return res;
+      }
+
+      /// [CRBA]  MatrixBase operator* (Constraint::Transpose S, ForceSet::Block)
+      template<typename Derived>
+      Eigen::Matrix<
+        Scalar,
+        3,
+        Derived::ColsAtCompileTime,
+        Options | Eigen::RowMajor,
+        3,
+        Derived::MaxColsAtCompileTime>
+      operator*(const Eigen::MatrixBase<Derived> & F) const
+      {
+        EIGEN_STATIC_ASSERT(
+          Derived::RowsAtCompileTime == 6, THIS_METHOD_IS_ONLY_FOR_MATRICES_OF_A_SPECIFIC_SIZE)
+        typedef Eigen::Matrix<
+          Scalar, 3, Derived::ColsAtCompileTime, Options | Eigen::RowMajor, 3,
+          Derived::MaxColsAtCompileTime>
+          ReturnType;
+
+        ReturnType res(3, F.cols());
+
+        // First two rows: S[:, 0:2]^T * F
+        res.template topRows<2>().noalias() =
+          ref.S.template leftCols<2>().transpose() * F.derived();
+
+        // Third row: [0,0,0,0,0,1]^T · F = F.row(5)
+        res.row(2) = F.row(5);
+
+        return res;
+      }
+    }; // struct TransposeConst
+
+    TransposeConst transpose() const
+    {
+      return TransposeConst(*this);
+    }
+
+    /* CRBA joint operators
+     *   - ForceSet::Block = ForceSet
+     *   - ForceSet operator* (Inertia Y,Constraint S)
+     *   - MatrixBase operator* (Constraint::Transpose S, ForceSet::Block)
+     *   - SE3::act(ForceSet::Block)
+     */
+    DenseBase matrix_impl() const
+    {
+      DenseBase res;
+
+      // Columns 0-1: copy the first two columns of S (6x2)
+      res.template leftCols<2>() = S.template leftCols<2>();
+
+      // Column 2: sparse [0, 0, 0, 0, 0, 1]^T
+      res.template block<5, 1>(0, 2).setZero();
+      res(5, 2) = Scalar(1);
+
+      return res;
+    }
+
+    template<typename MotionDerived>
+    typename MotionAlgebraAction<JointMotionSubspaceEllipsoidTpl, MotionDerived>::ReturnType
+    motionAction(const MotionDense<MotionDerived> & m) const
+    {
+      typedef
+        typename MotionAlgebraAction<JointMotionSubspaceEllipsoidTpl, MotionDerived>::ReturnType
+          ReturnType;
+      ReturnType res;
+
+      // Motion action on first two columns
+      motionSet::motionAction(m, S.template leftCols<2>(), res.template leftCols<2>());
+
+      // Motion action on third column [0, 0, 0, 0, 0, 1]^T
+      // [v; w] × [0; e_z] = [v × e_z; w × e_z]
+      const typename MotionDense<MotionDerived>::ConstLinearType & v_lin = m.linear();
+      const typename MotionDense<MotionDerived>::ConstAngularType & w = m.angular();
+
+      // v × e_z = [v[1], -v[0], 0]
+      res(LINEAR + 0, 2) = v_lin[1];
+      res(LINEAR + 1, 2) = -v_lin[0];
+      res(LINEAR + 2, 2) = Scalar(0);
+
+      // w × e_z = [w[1], -w[0], 0]
+      res(ANGULAR + 0, 2) = w[1];
+      res(ANGULAR + 1, 2) = -w[0];
+      res(ANGULAR + 2, 2) = Scalar(0);
```

Comment:
```suggestion
      res.col(2).noalias() = m.cross(SpatialAxis<5>());
```

This should do the same thing.
Like in revolute Joint you can create an Axis typedef: `typedef SpatialAxis<5> Axis`.

Reply:
Done. with an extra.